.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+# adapted from xarray's ci
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  detect-skip-ci-trigger:
+    name: "Detect CI Trigger: [skip-ci]"
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    outputs:
+      triggered: ${{ steps.detect-trigger.outputs.trigger-found }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+    - uses: keewis/ci-trigger@v1
+      id: detect-trigger
+      with:
+        keyword: "[skip-ci]"
+
+  ci:
+    name: py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    needs: detect-skip-ci-trigger
+
+    if: |
+      always()
+      && github.repository == 'xarray-contrib/pint-xarray'
+      && (
+        github.event_name == 'workflow_dispatch'
+        || needs.detect-skip-ci-trigger.outputs.triggered == 'false'
+      )
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+
+    steps:
+    - name: cancel previous runs
+      uses: styfle/cancel-workflow-action@0.6.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: checkout the repository
+      uses: actions/checkout@v2
+      with:
+        # need to fetch all tags to get a correct version
+        fetch-depth: 0  # fetch all branches and tags
+
+    - name: cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-py${{ matrix.python-version }}
+        restore-keys: |
+          pip-
+
+    - name: setup python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: upgrade pip
+      run: python -m pip install --upgrade pip setuptools wheel
+
+    - name: install dependencies
+      run: |
+        python -m pip install -r ci/requirements.txt
+
+    - name: install pint-xarray
+      run: python -m pip install --no-deps .
+
+    - name: show versions
+      run: python -m pip list
+
+    - name: import pint-xarray
+      run: |
+        python -c 'import pint_xarray'
+
+    - name: run tests
+      if: success()
+      id: status
+      run: |
+        python -m pytest --cov=pint_xarray --cov-report=xml
+
+    - name: Upload code coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: ./coverage.xml
+        flags: unittests
+        env_vars: PYTHON_VERSION
+        name: codecov-umbrella
+        fail_ci_if_error: false
```

Comment:
not sure if everything works as intended, but the commits are visible on codecov so I guess that's close enough?

Reply:
if not, I guess we can always open a new PR to fix that